aws/s3_test.go

```diff
@@ -132,332 +163,277 @@ func (b TestS3Bucket) addObject(s SetupInfo, fileName string, fileBody string) e
-		Region: awsgo.String(region)},
-	)
-	return session, err
+// TestListS3BucketArgs represents arguments for TestListS3Bucket
+type TestListS3BucketArgs struct {
+	bucketTags  []map[string]string
+	batchSize   int
+	shouldError bool
+	shouldMatch bool
 }
 
-// setupNukeTests sets up common operations for nuke S3 tests.
-func setupNukeTests(t *testing.T) SetupInfo {
-	var s SetupInfo
+// genTestList3Bucket - generates the test function for TestNukeS3Bucket
+func genTestListS3Bucket(args TestListS3BucketArgs) (func(t *testing.T), error) {
+	awsParams, err := setAWSParams()
 
-	region, err := getRandomRegion()
 	if err != nil {
-		assert.Fail(t, errors.WithStackTrace(err).Error())
+		return nil, fmt.Errorf("Failed to setup AWS params - %s", err.Error())
 	}
-	s.region = region
 
-	s.awsSession, err = createSession(region)
-	if err != nil {
-		assert.Fail(t, errors.WithStackTrace(err).Error())
-	}
-
-	s.svc = s3.New(s.awsSession)
-	if err != nil {
-		assert.Fail(t, errors.WithStackTrace(err).Error())
+	bucket := TestS3Bucket{
+		name: genTestBucketName(),
 	}
 
-	s.bucketName = genTestBucketName()
-	return s
-}
-
-func testListS3BucketsWrapper(t *testing.T, bucketTags []map[string]string, batchSize int, shouldMatch bool) {
-	s := setupNukeTests(t)
-
-	// Even if we nuke the bucket during our test - this will serve as a test to nuke non-existent bucket
-	// + also as our cleanup call
-
-	// Please note that we are passing the same session that was used to create the bucket
-	// This is required so that the defer cleanup call always gets the right bucket region
-	// to delete
-	defer nukeAllS3Buckets(s.awsSession, []*string{aws.String(s.bucketName)}, 1000)
-
 	awsSession, err := createSession("")
 	if err != nil {
-		assert.Failf(t, "Failed to create random session", errors.WithStackTrace(err).Error())
-	}
+		return nil, fmt.Errorf("Failed to create random session - %s", err.Error())
+	}
+
+	targetRegions := []string{awsParams.region}
+
+	return func(t *testing.T) {
+		// Please note that we are passing the same session that was used to create the bucket
+		// This is required so that the defer cleanup call always gets the right bucket region
+		// to delete
+		defer nukeAllS3Buckets(awsParams.awsSession, []*string{aws.String(bucket.name)}, 1000)
+
+		// Verify that - before creating bucket - it should not exist
+		//
+		// Please note that we are not reusing awsParams.awsSession and creating a random session in a region other
+		// than the one in which the bucket is created - this is useful to test the sceanrio where the user has
+		// AWS_DEFAULT_REGION set to region x but the bucket is in region y.
+		bucketNamesPerRegion, err := getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour*-1), targetRegions, bucket.name, args.batchSize)
+		if args.shouldError {
+			if err == nil {
+				assert.Fail(t, "Did not fail for invalid batch size")
+			}
+			t.Logf("SUCCESS: Did not list buckets due to invalid batch size - %s", bucket.name)
```

Comment:
We should not be using `t.Logf` for logging.

Reply:
Done.